FuncGodot Manual/pages/tips_runtime_building.html

```diff
@@ -0,0 +1,134 @@
+        <p>Once your map file path has been set, you can run the <code>verify_and_build()</code> method from the <a href="ref_func_godot_map.html" target="main">FuncGodotMap</a> node. When this is done, your map will be built just as it would have been by pressing the <i>Build</i> button in the Editor, along with all of the entity callbacks. Be aware that the <code>Engine.is_editor_hint()</code> conditional will return as false, in case you use it to control properties handling through the FuncGodot entity callbacks.</p>
+        <p>When the map builds, it will emit the signals <code>build_complete</code> or <code>build_failed</code>. You can use these signals to handle errors gracefully handle errors gracefully and to run any post-build code you need to run.</p>
+        <p>If using <a href="https://docs.godotengine.org/en/4.2/tutorials/3d/global_illumination/using_lightmap_gi.html" target="_blank">LightmapGI</a> for your global illumination solution, you can use the <i>FuncGodotMap</i> node's <code>unwrap_uv2()</code> method after receiving the <code>build_complete</code> signal. Once unwrapping has finished, the <i>FuncGodotMap</i> node will emit <code>unwrap_uv2_complete</code>.</p>
+        <blockquote><i>NOTE: Godot 4.2 does not currently support runtime lightmap baking or UV2 unwrapping in exported builds. For now, consider using a different lighting solution for runtime building.<br></i></blockquote>
+        An example of runtime map building is as follows:</p>
+        <pre><code>extends Node
+
+@onready var func_godot_map := $FuncGodotMap as FuncGodotMap
+
+# We can connect the various build signals of the FuncGodotMap 
+# in order to add more functionality after building.
+func _ready():
+    func_godot_map.connect("build_complete", _build_complete)
+    func_godot_map.connect("build_failed", _build_failed)
+    func_godot_map.connect("unwrap_uv2_complete", _unwrap_uv2_complete)
+    func_godot_map.local_map_file = "res://maps/runtime_building.map"
+    func_godot_map.verify_and_build()
+
+# We need to wait for the build to finish before unwrapping mesh UV2s for lightmap baking.
+# Currently lightmap baking cannot be performed at runtime, so it's not terribly useful yet.
+# Consider creating a procedural Voxel GI solid class entity that can be baked at runtime instead.
+func _build_complete() -> void:
+    print("Success! Unwrapping UV2...")
+    func_godot_map.unwrap_uv2()
+
+func _build_failed() -> void:
+    printerr("Failed to build the map file! :(")
+
+func _unwrap_uv2_complete() -> void:
+    print("UV2 unwrapping completed!")</code></pre>
+    <p>An example of this can be found in <a href="https://github.com/func-godot/func_godot_runtime_building" target="_blank">func_godot_runtime_building repository</a>.</p>
+
+    <h1>Building from Map Editors</h1>
+    <p>
+        Much like idTech engine games, it's possible to build and launch directly into a map from NetRadiant Custom and TrenchBroom alike.
```

Comment:
Other map editors are likely capable of this too, we just haven't tested them yet.

```html
<p>Much like idTech engine games, it's possible to build and launch directly into a map from some map editors. This can be a helpful tool for testing rapid iterations.</p>
```